protocol/app/upgrades/v5.0.0/upgrade.go

```diff
@@ -204,6 +206,39 @@ func initializeOIMFCaps(
 	}
 }
 
+func voteExtensionsUpgrade(
+	ctx sdk.Context,
+	keeper consensusparamkeeper.Keeper,
+) {
+	currentParams, err := keeper.Params(ctx, &consensustypes.QueryParamsRequest{})
+	if err != nil {
+		panic(fmt.Sprintf("failed to retrieve existing consensus params in VE upgrade handler: %s", err))
+	}
+	if currentParams.Params.Abci.VoteExtensionsEnableHeight != 0 {
+		panic(fmt.Sprintf(
+			"unable to update VE Enable Height since its current value of %d is already non-zero",
+			currentParams.Params.Abci.VoteExtensionsEnableHeight))
+	}
+	veEnableHeight := ctx.BlockHeight() + 1
+	currentParams.Params.Abci.VoteExtensionsEnableHeight = veEnableHeight
+	_, err = keeper.UpdateParams(ctx, &consensustypes.MsgUpdateParams{
+		Authority: keeper.GetAuthority(),
+		Block:     currentParams.Params.Block,
+		Evidence:  currentParams.Params.Evidence,
+		Validator: currentParams.Params.Validator,
+		Abci:      currentParams.Params.Abci,
+	})
+	if err != nil {
+		panic(fmt.Sprintf("failed to update consensus params with VE Enable Height %d: %s", veEnableHeight, err))
+	}
+	ctx.Logger().Info(
+		fmt.Sprintf(
+			"Successfully set VoteExtensionsEnableHeight to %d\n",
+			veEnableHeight,
+		),
+	)
+}
+
 // Initialize open interest for perpetuals
 func initializePerpOpenInterest(
```

Comment:
Is this the upgrade-handler entry-point? I'd prefer that we explicitly say this is upgrading consensus-params